genReadme.js

```diff
@@ -0,0 +1,288 @@
+    content: decoded,
+    sha: res.data.sha,
+  };
+}
+(async function main() {
+  try {
+    let readme = await getReadme();
+    default_options["from"] = readme;
+    let initialContent = readme.content;
+    initialContent = await updateSectionWith({
+      name: "setup",
+      path: "docs/basic/setup.md",
+      to: initialContent,
+      withToc: true,
+      headingLevel: 1,
+      prefix: "Section 1: ",
+    });
+    initialContent = await updateSectionWith({
+      name: "function-components",
+      path: "docs/basic/getting-started/function-components.md",
+      to: initialContent,
+    });
+    initialContent = await updateSectionWith({
+      name: "hooks",
+      path: "docs/basic/getting-started/hooks.md",
+      to: initialContent,
+    });
+    initialContent = await updateSectionWith({
+      name: "class-components",
+      path: "docs/basic/getting-started/class-components.md",
+      to: initialContent,
+    });
+    initialContent = await updateSectionWith({
+      name: "default-props",
+      path: "docs/basic/getting-started/default-props.md",
+      to: initialContent,
+      showHeading: false,
+    });
+    initialContent = await updateSectionWith({
+      name: "type-or-interface",
+      path: "docs/basic/getting-started/type-or-inteface.md",
+      to: initialContent,
+    });
+    initialContent = await updateSectionWith({
+      name: "basic-type-examples",
+      path: "docs/basic/getting-started/basic-type-examples.md",
+      to: initialContent,
+    });
+    initialContent = await updateSectionWith({
+      name: "react-prop-type-examples",
+      path: "docs/basic/getting-started/react-prop-type-examples.md",
+      to: initialContent,
+    });
+    initialContent = await updateSectionWith({
+      name: "forms-and-events",
+      path: "docs/basic/getting-started/forms-and-events.md",
+      to: initialContent,
+    });
+    initialContent = await updateSectionWith({
+      name: "context",
+      path: "docs/basic/getting-started/context.md",
+      to: initialContent,
+    });
+    initialContent = await updateSectionWith({
+      name: "forward-create-ref",
+      path: "docs/basic/getting-started/forward-create-ref.md",
+      to: initialContent,
+    });
+    initialContent = await updateSectionWith({
+      name: "portals",
+      path: "docs/basic/getting-started/portals.md",
+      to: initialContent,
+    });
+    initialContent = await updateSectionWith({
+      name: "context",
+      path: "docs/basic/getting-started/error-boundaries.md",
+      to: initialContent,
+    });
+    initialContent = await updateSectionWith({
+      name: "concurrent",
+      path: "docs/basic/getting-started/concurrent.md",
+      to: initialContent,
+    });
+    initialContent = await updateSectionWith({
+      name: "types",
+      path: "docs/basic/troubleshooting/types.md",
+      to: initialContent,
+      withToc: true,
+      headingLevel: 1,
+    });
+    initialContent = await updateSectionWith({
```

Comment:
i think you forgot to add sections for Context and gDSFP and maybe one more

Reply:
yikes! I'll look into it.

Reply:
Found it! I replaced **context** with error boundaries and didn't generate **error boundaries** correctly. So these two were missing. Also, mistook default props for **derived state from props** was also missing, just like you said!

These would not be a problem if the paths are dynamically aggregated before update but that's a PR for another day.